internal/cluster/distribution/eks/ekscluster/nodepools/wait_for_asg.go

```diff
@@ -80,7 +81,24 @@ func WaitForASGToBeFulfilled(
 					return nil
 				}
 			} else {
-				return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s", waitAttempts, waitInterval)
+				asGroup, err := m.GetAutoscalingGroupByStackName(asgName)
+				if err != nil {
+					return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s, could not get ASG in stack: %s", waitAttempts, waitInterval, asgName)
+				}
+
+				lastActivity, _ := m.LastAutoscalingActivity(ctx, asGroup.AutoScalingGroupName)
+				if err != nil {
+					return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s, could not get last ASG activity in stack: %s", waitAttempts, waitInterval, asgName)
+				}
```

Comment:
**Required**: you forgot to reassign the err variable before the check.

```suggestion
				lastActivity, err := m.LastAutoscalingActivity(ctx, asGroup.AutoScalingGroupName)
				if err != nil {
					return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s, could not get last ASG activity in stack: %s", waitAttempts, waitInterval, asgName)
				}
```

Reply:
Oh my! Good catch, thanks!